pkg/fakerp/config.go

```diff
@@ -54,3 +61,82 @@ func NewConfig() (*Config, error) {
 	}
 	return &c, nil
 }
+
+func (s *Server) configure() (*api.PluginConfig, error) {
+	tc := api.TestConfig{
+		RunningUnderTest:      os.Getenv("RUNNING_UNDER_TEST") != "",
+		ImageResourceGroup:    os.Getenv("IMAGE_RESOURCEGROUP"),
+		ImageResourceName:     os.Getenv("IMAGE_RESOURCENAME"),
+		DeployOS:              os.Getenv("DEPLOY_OS"),
+		ImageOffer:            os.Getenv("IMAGE_OFFER"),
+		ImageVersion:          os.Getenv("IMAGE_VERSION"),
+		ORegURL:               os.Getenv("OREG_URL"),
+		EtcdBackupImage:       os.Getenv("ETCDBACKUP_IMAGE"),
+		AzureControllersImage: os.Getenv("AZURE_CONTROLLERS_IMAGE"),
+	}
+
+	// populate azure artifacts
+	var cb, kb, db []byte
+	var crt *x509.Certificate
+	var key *rsa.PrivateKey
+	var err error
+
+	artifactDir := "./test/artifacts/azure/"
+	// if test files exist, load them
+	if fileExist(artifactDir+"osa-int-infra.cert") &&
+		fileExist(artifactDir+"osa-int-infra.key") {
+		s.log.Debug("read azure artifacts from artifacts")
+		cb, err = ioutil.ReadFile(artifactDir + "osa-int-infra.cert")
+		if err != nil {
+			return nil, err
+		}
+		kb, err = ioutil.ReadFile(artifactDir + "osa-int-infra.key")
+		if err != nil {
+			return nil, err
+		}
+		crt, err = tls.ParseCert(cb)
+		if err != nil {
+			return nil, err
+		}
+		key, err = tls.ParsePrivateKey(kb)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		s.log.Debug("generate dummy azure certificates")
+		key, crt, err = tls.NewCert(&tls.CertParams{})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// TODO: We should remove or re-publish these images to avoid secret
```

Comment:
If we can remove the need for this secret either by republishing images or MSFT opening them, we will not need to distribute "team secret" around